docs/howto/use_pum_for_migrations.rst

```diff
@@ -0,0 +1,95 @@
+Use Pum for schema migrations
+-----------------------------
+
+When Procrastinate developers make changes to the Procrastinate database schema they
+write migration scripts.
+
+Here's an example of a migration script:
+
+.. code-block:: sql
+
+    ALTER TABLE procrastinate_jobs ADD COLUMN extra TEXT;
+
+Migration scripts are pure-SQL scripts, meaning that they may be applied to the database
+using any PostgreSQL client, including ``psql`` and ``PGAdmin``.
+
+The names of migration script files adhere to a certain pattern, which comes from the
+`Pum`_ project. Pum is thus a good choice for Procrastinate migrations, although other
+migration tools may be used as well.
+
+See the `Pum README`_ on GitHub to know how to use Pum. But below is a Pum quickstart,
+applied to Procrastinate.
+
+First of all, note that Pum connects to the PostgreSQL database using a `PostgreSQL
+Connection Service`_. So you have to have a Connection Service File. Here's an example
+of a Connection Service File (located at ``~/.pg_service.conf``):
+
+.. code-block::
+
+    [procrastinate]
+    host=localhost
+    port=5432
+    user=postgres
+    password=xxxxxxx
+    dbname=procrastinate
+
+You can use ``psql service=procrastinate`` to check that you can connect to
+PostgreSQL through the ``procrastinate`` service defined in your Connection Service
+File.
+
+The first time you install Procrastinate you will need to apply the Procrastinate schema
+to the database using the ``procrastinate schema --apply`` command.
+
+If you decide to use Pum for handling migrations you'll also need to create the Pum
+table (which Pum uses internally for tracking schema versions), and set the ``baseline``
+in this table.
+
+Let's say you are installing Procrastinate 0.5.0. Your baseline will then be 0.5.0. This
+the Pum command to set the baseline:
+
+.. code-block:: console
+
+    pum baseline -p procrastinate -t public.pum -d procrastinate/sql/migrations/ -b 0.5.0
```

Comment:
Can we use long form flags ? (maybe have both lines with long and short forms)

Reply:
The path will not be `procrastinate/sql/migrations/` in the case of people not installing procrastinate from source. We may need to export a dedicated entrypoint:
```
procrastinate schema --migrations-path
```
?

Reply:
> Can we use long form flags ? (maybe have both lines with long and short forms)

Yep. The long form is now used.

Reply:
> The path will not be procrastinate/sql/migrations/ in the case of people not installing procrastinate from source.

Good call! I've added new commits for that.

Would you prefer a separate PR for the new `schema --migrations-path` command?

Reply:
no it's good like this.